### PR TITLE
chore: bump toolchain to v4.3.0

### PR DESCRIPTION
**Pull request by semorrison**
With completion approaching, it seems like it would nice to bump the mathlib dependency up to a round number --- this moves Mathlib up to the version tagged `v4.3.0`, i.e. the designated version compatible with the `v4.3.0` stable release of Lean.

### CODE REVIEW

File: lake-manifest.json
Comment (YaelDillies) [inline lake-manifest.json:1]: Why all these changes?

**Comment by teorth** [comment]
The last time we tried to update mathlib https://github.com/teorth/pfr/pull/26 it broke a lot of things and multiple contributors had to help clean up the resulting mess.  I assume that you're not doing whatever was wrong in that previous PR?

File: lake-manifest.json
Comment (semorrison) [reply lake-manifest.json:1]: `lake build` worked?

Comment (semorrison) [reply lake-manifest.json:1]: I've made the diff look smaller by ensuring that json blocks aren't reordered. I've update mathlib and its upstream dependencies, but not touched docgen or its dependencies.

Comment (YaelDillies) [reply lake-manifest.json:1]: You can see that `lake build` is not what we run to build the project in CI. The relevant lines are here: https://github.com/teorth/pfr/blob/master/.github/workflows/push.yml#L43C10-L43C50

**Comment by teorth** [comment]
OK, now that we have at least one working build to revert to, I'm willing to risk the mathlib upgrade, but I'll wait for Yael to sign off too as he has more experience with this sort of thing.

**Comment by YaelDillies** [comment]
I think it's fine to merge this to have one nice stable commit to point to.

However the project won't be pinned on 4.3.0 forever because we will bump it to the newest mathlib every time a PR for preliminary material gets merged.

**Pull request merged**